### PR TITLE
Clean up STs

### PR DESCRIPTION
**Pull request by tomdee**

### CODE REVIEW

**Comment by tomdee** [comment]
@robbrockbank This is ready for you to review. The tests are passing locally but I can't test them on Semaphore without updating the docker version (which will break tests on master)

File: Makefile [hunk 1]
```diff
@@ -26,6 +26,11 @@ calicobuild.created: $(BUILD_FILES)
 	docker build -f build_calicoctl/Dockerfile -t calico/build .
 	touch calicobuild.created
 
+docker:
```
Comment (robbrockbank) [inline Makefile]: make clean should remove this binary

Comment (robbrockbank) [inline Makefile]: Also need to add any of these new downloaded files to .gitignore

Comment (robbrockbank) [inline Makefile]: *personally* i dislike filling up the root of the project with cruddy un-checked in files.  I'd rather we had a single directory reserved for these kind of downloaded files.  but as long as .gitignore is updated I'm not overly worried.

File: calico_containers/tests/st/utils/network.py [hunk 2]
```diff
@@ -25,9 +27,10 @@ class DockerNetwork(object):
     def __init__(self, host, name, driver="calico"):
         """
         Create the network.
-        :param host: The Docker Host which creates the network.
+        :param host: The Docker Host which creates the network (note that
+        networks
         :param name: The name of the network.  This must be unique per cluster
-        and is the user-facing identifier for the network.  (Calico itself will
+        and it the user-facing identifier for the network.  (Calico itself will
```
Comment (robbrockbank) [inline calico_containers/tests/st/utils/network.py]: i think this should still be  "is"

File: calico_containers/tests/st/utils/utils.py [hunk 3]
```diff
@@ -113,3 +130,83 @@ def check_bird_status(host, expected):
...
+def assert_number_endpoints(host, expected):
+    """
+    Check that a host has the expected number of endpoints in Calico
+    Parses the "calicoctl endpoint show" command for number of endpoints.
+    Raises AssertionError if the number of endpoints does not match the
+    expected value.
+
+    :param host: DockerHost object
+    :param expected: int, number of expected endpoints
+    :return:
```
Comment (robbrockbank) [inline calico_containers/tests/st/utils/utils.py]: return None

**Comment by robbrockbank** [comment]
Minor comments and the LGTM once tests completed=.  Don't need to see again unless you make some major changes.

**Pull request merged**